internal/resources/ekscluster/data_source.go

```diff
@@ -79,6 +82,36 @@ func dataSourceTMCEKSClusterRead(ctx context.Context, d *schema.ResourceData, m
 			return true, nil
 		}
 
+		if isWaitForKubeconfig(d) {
+			clusFullName := &clustermodel.VmwareTanzuManageV1alpha1ClusterFullName{Name: resp.EksCluster.Spec.AgentName, OrgID: clusterFn.OrgID, ManagementClusterName: "eks", ProvisionerName: "eks"}
+			clusterResp, err := config.TMCConnection.ClusterResourceService.ManageV1alpha1ClusterResourceServiceGet(clusFullName)
+			// nolint: wsl
+			if err != nil {
+				log.Printf("Unable to get Tanzu Mission Control cluster entry, name : %s", clusterFn.Name)
+				return true, nil
+			}
+
+			if !isManagemetClusterHealthy(clusterResp) {
+				log.Printf("[DEBUG] waiting for cluster(%s) to be in Healthy status", clusterFn.Name)
+				return true, nil
+			}
+
+			fn := &configModels.VmwareTanzuManageV1alpha1ClusterFullName{
+				ManagementClusterName: "eks",
+				ProvisionerName:       "eks",
+				Name:                  resp.EksCluster.Spec.AgentName,
+				OrgID:                 "",
+			}
+			resp, err := config.TMCConnection.KubeConfigResourceService.KubeconfigServiceGet(fn)
+
+			if kubeConfigReady(err, resp) {
+				if err = d.Set(kubeconfigKey, resp.Kubeconfig); err != nil {
+					log.Printf("Failed to set Kubeconfig")
+					return true, nil
```

Comment:
If setting the kubeconfig fails, we don't need to retry, it's probably non-recoverable error.

Reply:
You also need to return the error so it's propagated up.

Reply:
@asutoshpalai  fixed it